File: 2015/Day_03/Python/test_part_2.py
from . import part_2


def test_separate_directions():
    assert part_2.separate_directions('>') == ['>']
    assert part_2.separate_directions('^>v<') == ['^', '>', 'v', '<']
    assert part_2.separate_directions('^v^v^v^v^v') == ['^', 'v', '^', 'v', '^', 'v', '^', 'v', '^', 'v']


def test_count_house_visits():
    assert part_2.count_houses_visited(['^', "v"]) == 3
    assert part_2.count_houses_visited(['^', '>', 'v', '<']) == 3
    assert part_2.count_houses_visited(['^', 'v', '^', 'v', '^', 'v', '^', 'v', '^', 'v']) == 11

File: 2015/Day_03/Ruby/test_part_2.rb
require_relative "part_2"
require 'test/unit'


class TestSantaRoute < Test::Unit::TestCase
    
    def test_separate_directions
        assert_equal(['>'], separate_directions('>'))
        assert_equal(['^', '>', 'v', '<'], separate_directions("^>v<"))
    end
    
    def test_count_houses_visited
        assert_equal(3, count_houses_visited(["^", "v"]))
    end
end
